extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -357,6 +358,27 @@ public List<String> responseHeader(final HttpRequest request, final HttpResponse
         }
     }
 
+    private static class ClientSpanNameExtractor implements SpanNameExtractor<HttpRequest> {
+
+        @Override
+        public String extract(HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpRequestHead) {
+                HttpRequestHead head = (HttpRequestHead) httpRequest;
+                if (head.traceOperation != null) {
+                    return head.traceOperation;
+                }
+            }
+            if (httpRequest instanceof WriteHeadersHttpRequest) {
+                WriteHeadersHttpRequest writeHeaders = (WriteHeadersHttpRequest) httpRequest;
+                String traceOperation = writeHeaders.traceOperation();
+                if (traceOperation != null) {
+                    return traceOperation;
+                }
+            }
+            return "HTTP " + httpRequest.method().name();
```

Comment:
This should delegate back to the `HttpSpanNameExtractor`.

Reply:
Will do.

Reply:
done